sklearn/sklearn-template/template/README.md

```diff
@@ -0,0 +1,105 @@
+Cloud ML Engine (CMLE) at scale. Before you jump in, let’s cover some of the different tools you’ll be using to get 
+your job running on CMLE.
+
+- [Google Cloud Platform](https://cloud.google.com/) lets you build and host applications and websites, store data, 
+and analyze data on Google's scalable infrastructure.
+- [Cloud Machine Learning Engine](https://cloud.google.com/ml-engine/) is a managed service that enables you to easily build 
+machine learning models that work on any type of data, of any size.
+- [Google Cloud Storage](https://cloud.google.com/storage/) (GCS) is a unified object storage for developers and 
+enterprises, from live data serving to data analytics/ML to data archiving.
+- [Cloud SDK](https://cloud.google.com/sdk/) is a command line tool which allows you to interact with 
+Google Cloud products.
+- [Google BigQuery](https://cloud.google.com/bigquery/) A fast, highly scalable, cost-effective, and fully managed 
+cloud data warehouse for analytics, with even built-in machine learning.
+
+# Structure of the template
+```
+Template              
+    |__ scripts
+        |__ train.sh            # convenience script for running machine learning training jobs 
+    |__ trainer                 # trainer package
+        |__ metadata.py         # dataset metadata and feature columns definitions
+        |__ model.py            # pre-processing and machine learning model pipeline definition
+        |__ utils.py            # utility functions including e.g. loading data from bigquery and cloud storage
+        |__ task.py             # training job entry point, handling the parameters passed from command line 
+    |__ hptuning_config.yaml    # config file for running hyperparameter tunning 
+    |__ setup.py                # specify necessary dependency of the training job
+    |__ requirements.txt        # specify necessary dependency of the training job
+    |__ Dockerfile
+```
+
+# Steps to make use of the template
+## Step 0. Prerequisite
+Before you follow the instructions below to adapt the tempate to your machine learning job, 
+you need a Google cloud project if you don't have one. You can find detailed instructions 
+[here](https://cloud.google.com/dataproc/docs/guides/setup-project).
+
+- Make sure the following API & Services are enabled.
+    * Cloud Storage
+    * Cloud Machine Learning Engine
+    * BigQuery API
+    * Cloud Build API (for CI/CD integration)
+    * Cloud Source Repositories API (for CI/CD integration)
+
+- Configure project id as environment variable
+  ```bash
+  $ export PROJECT_ID=[your-google-project-id]
+  ```
+  
+## Step 1. Modify metadata.py
+```python
+# Example for iris dataset
+COLUMNS = None  # Schema of the data. Necessary for data stored in GCS
+
+NUMERIC_FEATURES = [
+    'sepal_length',
+    'sepal_width',
+    'petal_length',
+    'petal_width',
+]
+
+CATEGORICAL_FEATURES = [
+
```

Comment:
Since we are using Iris, this doesn't need to be filled. But I think we need to give a comment to fill this with any categorical features in the dataset.

Reply:
Added.